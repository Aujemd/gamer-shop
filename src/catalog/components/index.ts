export { default as CatalogHeader } from "./catalog-header";
