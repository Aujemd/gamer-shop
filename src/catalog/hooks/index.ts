export { useCatalogProducts } from "./use-catalog-products";
