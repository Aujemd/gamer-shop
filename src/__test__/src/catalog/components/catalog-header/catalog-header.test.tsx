import { render, screen } from "@testing-library/react";

import { CatalogHeader } from "@/catalog";

describe("catalogHeader Component", () => {
    it("should renders correctly", () => {
        const { asFragment } = render(<CatalogHeader data-testid="catalog-header-section-test-id" />);
        expect(screen.getByTestId("catalog-header-section-test-id")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the page title", () => {
        render(<CatalogHeader />);
        const title = screen.getByTestId("catalog-header-title-test-id");
        expect(title).toBeInTheDocument();
        expect(title).toHaveTextContent(/top sellers/i);
    });

    it("renders the genre label", () => {
        render(<CatalogHeader />);
        const label = screen.getByTestId("catalog-genre-select-label-test-id");
        expect(label).toBeInTheDocument();
        expect(label).toHaveTextContent(/genre/i);
        expect(label).toHaveAttribute("for", "catalog-genre-select");
    });

    it("renders the select with options", () => {
        render(<CatalogHeader />);
        const select = screen.getByTestId("catalog-genre-select-test-id");
        expect(select).toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(<CatalogHeader className="custom-header" data-testid="catalog-header-section" />);
        expect(screen.getByTestId("catalog-header-section")).toHaveClass("custom-header");
    });
});
