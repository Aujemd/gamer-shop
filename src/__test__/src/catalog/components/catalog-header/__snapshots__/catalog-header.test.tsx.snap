// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalogHeader Component should renders correctly 1`] = `
<DocumentFragment>
  <section
    class="py-8 px-6 border-b border-b-stroke-tertiary "
    data-testid="catalog-header-section-test-id"
  >
    <div
      class="max-w-desktop mx-auto"
    >
      <h1
        class="font-archivo font-bold text-2xl leading-7 tracking-sm text-center text-gray-medium lg:text-4xl lg:leading-10 text-left uppercase lg:normal-case"
        data-testid="catalog-header-title-test-id"
      >
        Top Sellers
      </h1>
      <div
        class="mt-8 lg:mt-12 divide-stroke-secondary divide-x lg:flex lg:items-center lg:justify-end"
      >
        <label
          class="font-archivo font-bold text-xl leading-6 tracking-sm text-center pr-6"
          data-testid="catalog-genre-select-label-test-id"
          for="catalog-genre-select"
        >
          Genre
        </label>
        <select
          class="w-select-custom-width font-archivo text-xl font-normal leading-6 tracking-sm text-left  py-4 pl-10"
          data-testid="catalog-genre-select-test-id"
          id="catalog-genre-select"
        >
          <option
            value="1"
          >
            Option 1
          </option>
          <option
            value="2"
          >
            Option 2
          </option>
          <option
            value="3"
          >
            Option 3
          </option>
        </select>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
